File: output/ChatGPT/codenet/C++/Java/s129088151.java
import java.util.HashMap;
import java.util.Scanner;

public class s129088151 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        scanner.nextLine();

        HashMap<String, Boolean> T = new HashMap<>();

        for (int i = 0; i < n; i++) {
            String[] input = scanner.nextLine().split(" ");
            String com = input[0];
            String str = input[1];
            if (com.equals("i")) {
                T.put(str, true);
            } else {
                if (T.containsKey(str)) {
                    System.out.println("yes");
                } else {
                    System.out.println("no");
                }
            }
        }
        scanner.close();
    }
}

File: output/CodeGen/codenet/C++/Python/s129088151.py
def main():
    n = int(input())
    T = {}
    for i in range(n):
        com, str = input().split()
        if com[0] == 'i':
            T[str] = True
        else:
            if T[str]:
                print('yes')
            else:
                print('no')

if __name__ == '__main__':
    main()